system_metrics_collector/test/system_metrics_collector/test_linux_memory_measurement.cpp

```diff
@@ -143,37 +227,210 @@ class LinuxMemoryMeasurementTestFixture : public ::testing::Test
+
+    // check measurements
+    for (int i = 0; i < STATISTICS_DATA_TYPES.size(); ++i) {
+      EXPECT_EQ(STATISTICS_DATA_TYPES[i], msg.statistics[i].data_type);
+      if (std::isnan(expected_stats[times_received][i].data)) {
+        EXPECT_TRUE(std::isnan(msg.statistics[i].data));
+      } else {
+        EXPECT_DOUBLE_EQ(expected_stats[times_received][i].data, msg.statistics[i].data);
+      }
+    }
+
+    ++times_received;
+  }
+
+  rclcpp::Subscription<MetricsMessage>::SharedPtr subscription;
+  std::array<StatisticDataPoint[STATISTICS_DATA_TYPES.size()], 6> expected_stats;
+  int times_received;
+};
 
 TEST(LinuxMemoryMeasurementTest, testReadInvalidFile)
 {
   const auto s = system_metrics_collector::readFileToString("this_will_fail.txt");
   ASSERT_EQ("", s);
 }
 
-TEST(LinuxMemoryMeasurementTest, testProcessLines)
+TEST(LinuxMemoryMeasurementTest, testprocessMemInfoLines)
 {
-  auto d = system_metrics_collector::processMemInfoLines(EMPTY_SAMPLE);
+  auto d = processMemInfoLines(EMPTY_SAMPLE);
   ASSERT_TRUE(std::isnan(d));
 
-  d = system_metrics_collector::processMemInfoLines(GARBAGE_SAMPLE);
+  d = processMemInfoLines(GARBAGE_SAMPLE);
   ASSERT_TRUE(std::isnan(d));
 
-  d = system_metrics_collector::processMemInfoLines(INCOMPLETE_SAMPLE);
+  d = processMemInfoLines(INCOMPLETE_SAMPLE);
   ASSERT_TRUE(std::isnan(d));
 
-  d = system_metrics_collector::processMemInfoLines(COMPLETE_SAMPLE);
+  d = processMemInfoLines(COMPLETE_SAMPLE);
   ASSERT_DOUBLE_EQ(MEMORY_USED_PERCENTAGE, d);
 
-  d = system_metrics_collector::processMemInfoLines(FULL_SAMPLE);
+  d = processMemInfoLines(FULL_SAMPLE);
   ASSERT_DOUBLE_EQ(MEMORY_USED_PERCENTAGE, d);
 }
+
+TEST_F(LinuxMemoryMeasurementTestFixture, testManualMeasurement) {
+  test_measure_linux_memory->setTestString("");
+  double mem_used_percentage = test_measure_linux_memory->periodicMeasurement();
+  ASSERT_TRUE(std::isnan(mem_used_percentage));
+
+  test_measure_linux_memory->setTestString(FULL_SAMPLE);
+  mem_used_percentage = test_measure_linux_memory->periodicMeasurement();
+  ASSERT_DOUBLE_EQ(MEMORY_USED_PERCENTAGE, mem_used_percentage);
+}
+
+TEST_F(LinuxMemoryMeasurementTestFixture, testPublishMetricsMessage)
+{
+  ASSERT_NE(test_measure_linux_memory, nullptr);
+  ASSERT_FALSE(test_measure_linux_memory->isStarted());
+
+  auto test_receive_measurements = std::make_shared<TestReceiveMemoryMeasurementNode>(
+    "test_receive_measurements");
+  std::promise<bool> empty_promise;
+  std::shared_future<bool> dummy_future = empty_promise.get_future();
+  rclcpp::executors::SingleThreadedExecutor ex;
+  ex.add_node(test_measure_linux_memory);
+  ex.add_node(test_receive_measurements);
+
+  //
+  // spin the node with it started
+  //
+  bool start_success = test_measure_linux_memory->start();
+  ASSERT_TRUE(start_success);
+  ASSERT_TRUE(test_measure_linux_memory->isStarted());
+  ex.spin_until_future_complete(dummy_future, TEST_LENGTH);
```

Comment:
TEST_LENGTH is a bit confusing - TEST_DURATION

Reply:
Done.